### ISSUE TITLE
Accessibility: Status message not announced by screen readers [Product Single (Grouped)]

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#51929]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The error message if the Add to Cart button is selected without choosing products is visually displayed, but screen readers do not announce it (though it has a role of alert) because the page refreshes and screen readers instead read the page title/top of the page.

![Image](https://github.com/user-attachments/assets/7fdad3e1-2633-45ab-997f-50823f0bd65f)

### Component or Page: 

Page: Product Single (Grouped)  

### Name affected element:

Error message if Add to Cart button is selected without choosing products

### Relevant code snippets:

`<div class=""wc-block-components-notice-banner is-error"" role=""alert"">
	<svg xmlns=""http://www.w3.org/2000/svg"" viewBox=""0 0 24 24"" width=""24"" height=""24"" aria-hidden=""true"" focusable=""false"">
		<path d=""M12 3.2c-4.8 0-8.8 3.9-8.8 8.8 0 4.8 3.9 8.8 8.8 8.8 4.8 0 8.8-3.9 8.8-8.8 0-4.8-4-8.8-8.8-8.8zm0 16c-4 0-7.2-3.3-7.2-7.2C4.8 8 8 4.8 12 4.8s7.2 3.3 7.2 7.2c0 4-3.2 7.2-7.2 7.2zM11 17h2v-6h-2v6zm0-8h2V7h-2v2z""></path>
	</svg>
	<div class=""wc-block-components-notice-banner__content"">
		Please choose a product to add to your cart…	</div>
</div>`

### Recommended fix:

Possible options to remediate this:

A) Don't refresh the page on submit of the ""Add to Cart"" button and instead use JavaScript to triggert the action/show the error.

B) After the page is refreshed, use JavaScript to set keyboard focus on the error message so screen readers read it rather than the content at the top of the page.

**Comment by lanej0** [comment]
Duplicate of https://github.com/woocommerce/woocommerce/issues/51931 -- closing this and merging it into there.

**Comment by vikrampm1** [comment]
@lanej0 this is not a duplicate of #51931 as I mentioned in the description there that these are connected. I just updated the titles for both to differentiate, so please re-open this issue.

**Comment by vikrampm1** [comment]
@lanej0 / @mikejolley can you please help open this issue back as noted above?

**Comment by Manussakis** [comment]
I was not able to reproduce this issue. Most of the issues with alert messages were fixed once the https://github.com/woocommerce/woocommerce/pull/50061 PR was merged.
Please see the screenshots below:

- Error message being focused and screen reader announcing it after adding to cart without selecting the number of products:
![Image](https://github.com/user-attachments/assets/4ddd0bab-82e3-4b48-aa1b-cdec41ccd1b4)

- Success message being focused and screen reader announcing it after adding products to the cart with the quantity selected.
![Image](https://github.com/user-attachments/assets/f99703ad-1a2f-47a4-9f69-bf6b46b586fd)

I can see that the screenshot in the issue description shows the products with a checkbox instead of a quantity input on the left. Maybe this explains why I was not able to reproduce it. In that case, I will need an explanation of how to display the checkboxes since I couldn't reproduce it.

cc: @vikrampm1

**Comment by vikrampm1** [comment]
>In that case, I will need an explanation of how to display the checkboxes since I couldn't reproduce it.

@amberhinds can you able to help with the above?

**Comment by amberhinds** [comment]
We have a grouped product with checkboxes here: https://woo.equalizedigital.com/product/mediocre-granite-computer/ (password: wooa11y)

I have retested this with the current version of Woo on that page with both NVDA and VoiceOver and can confirm that the error message  is now being read out.

**Comment by vikrampm1** [comment]
Thanks @amberhinds!

**Comment by Manussakis** [comment]
@vikrampm1 I linked the https://github.com/woocommerce/woocommerce/pull/50061 PR to this issue.
